### ISSUE TITLE
Event received by organisation not in sharing group distribution (followup)

### ISSUE THREAD

**Issue by liz-cullen** [MISP/MISP#1721]
Following on #1720 

I added org2, org3 and org4 to M1 so that M2, M3, M4 could use a sync user in their organisation when syncing to M1 but get the same problem namely org4 on M1 and M4 can see events even though not in the sharing group.

org4 has been added to the sharing group at M4 but not M1

**Comment by liz-cullen** [comment]
BTW what role should the sync users on M1 have. I find if it is set to sync user or org admin then neither M3 not M4 can see sharing group events but if it is set to admin then both M3 and M4 can see them.

**Comment by iglocska** [comment]
Sync, otherwise you are giving the other instances admin access to your data.

**Comment by liz-cullen** [comment]
That's what I thought but when set this way the sync user at M1 for org2 can't see events pushed by org2 at M2 with a sharing group comprising org2(local) and org3(remote) and roaming enabled.

**Comment by iglocska** [comment]
Could you check the sharing group on M1 with an admin user and see what the distribution list is?

**Comment by liz-cullen** [comment]
The sharing group is listed on M1.

**Comment by liz-cullen** [comment]
Sorry I meant isn't listed

**Comment by iglocska** [comment]
Just to recap:

Org 2 on M2 creates an event with a sharing group that is set to:

instances: Roaming
orgs: Org 2, Org 3

Org 2 then pushes the event to M1? That shouldn't work in the first place since roaming mode requires the host organisation of M1 to be Org 2 or Org 3 for the event to be pushable. When setting up the sync link from M2 -> M1, which organisation did you pick as the host organisation?

**Comment by liz-cullen** [comment]
When setting up the sync link from M2 to M1  I set the external organisation to org2 - is that what you mean by host organisation.

**Comment by iglocska** [comment]
Yeah, that should be the host organisation of M1, so org1.

**Comment by liz-cullen** [comment]
So to summarise.

I should create a sync user of M1 in org1 and this user is used by M2, M3 and M4 when syncing with M1. 
The if I create a SG at M2 containing just org2 and org3 will events with that distribution be forwarded to M3 but M4?

**Comment by liz-cullen** [comment]
Also how does this relate to your previous comment #1720 

Basically the other misps log in as the sync users and fetch anything they
can see, if m4 has an org 4 sync user on m1 it should not be able to see
any sharing groups that don't include org 4

**Comment by iglocska** [comment]
OK just to recap it all how the setup should go:

M1 wants to interconnect with M2.
M1's host organisation (the organisation that runs M1) is org1.
M2's host organisation (the organisation that runs M2) is org2.

Org 1 creates a synchronisation user on M1 for Org 2 (so the user is a sync user of org 2)
Org 2 creates a synchronisation user on M2 for Org 1 (so the user is a sync user of org 1)

Org 1 adds a new synchronisation link on M1 to M2, defines the remote organisation as Org 2.
Org 2 adds a new synchronisation link on M2 to M1, defines the remote organisation as Org 1.

**Comment by liz-cullen** [comment]
Is it necessary to create a sync user / sync link for M1 on M2 if it is not going to pull/push data from M2?

**Comment by iglocska** [comment]
It is not enforced but there are some functionalities that an automation
user might miss out on. Also, my example above is for a 2 way
synchronisation, you can absolutely drop one side and let a one way link
handle pulling and pushing (though this comes with the downside that you
won't get real time information from one instance to the other, only at
regular intervals whenever you initiate a pull).

**Comment by liz-cullen** [comment]
I assume when you create a sync link you use the authentication key of the sync user for your org.

But when I do this events M2 with a sharing group distribution are still not being synced with M1.

**Comment by iglocska** [comment]
That is correct, so now that you have set this up:

Create an event on M2 that either includes org 1 in roaming mode or org 1 + M1 named. This should be synchronised with M1.

**Comment by liz-cullen** [comment]
Yes I did that (org 1 in roaming mode) but it is not synced with M1.

**Comment by iglocska** [comment]
Interesting, I'll try to test it locally, it should be synced. Is org2 also included?

**Comment by liz-cullen** [comment]
Yes
